pkgs/by-name/pa/paup/package.nix

```diff
@@ -0,0 +1,50 @@
+{
+  lib,
+  stdenvNoCC,
+  fetchurl,
+  autoPatchelfHook,
+  curl,
+  gfortran,
+  zlib,
+}:
+
+stdenvNoCC.mkDerivation {
+  pname = "paup";
+  version = "4.0a168";
+
+  src = fetchurl {
+    url = "http://phylosolutions.com/paup-test/paup4a168_centos64.gz";
+    hash = "sha256-41dZswlrIQ05f1zJzId78DKmPf0QH1SfrexzvCAUxq8=";
+  };
+
+  unpackPhase = ''
+    runHook preUnpack
+
+    gunzip -c $src > paup
+    chmod +x paup
+
+    runHook postUnpack
+  '';
+
+  nativeBuildInputs = [ autoPatchelfHook ];
+  buildInputs = [ curl gfortran zlib ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    mv paup $out/bin
```

Comment:
`install` command is useful instead of these operations and chmod.
```
install -Dm755 paup -t $out/bin
```

Reply:
Done.